packages/amplify-graphql-schema-generator/src/ts-schema-generator/helpers.ts

```diff
@@ -84,18 +94,49 @@ const getTypescriptDataSchemaType = (type: string): string => {
   return tsDataSchemaType ?? DEFAULT_DATATYPE;
 };
 
-const createModelDefinition = (model: Model): ts.Node => {
+const createModelDefinition = (model: Model, engine: string): ts.Node => {
   const properties = model.getFields().map((field) => {
-    return createProperty(field);
+    return createProperty(field, engine, model.getName());
   });
   return ts.factory.createObjectLiteralExpression(properties as ts.ObjectLiteralElementLike[], true);
 };
 
-const createModel = (model: Model): ts.Node => {
+/**
+ * Creates a typescript data schema type from internal SQL schema representation
+ * Example typescript data schema type output: `a.enum()`
+ * @param type SQL IR Enum type
+ * @returns Typescript data schema type in TS Node format
+ */
+const createEnums = (type: EnumType, engine: string, modelName: string, enumName: string): ts.Node => {
+  if (engine === 'Postgres'){
```

Comment:
nit: since `engine` is a string rather than a strongly typed value, let's make the engine handling in this method explicit and fail for unsupported engine types. That'll protect against typos in calling code. Alternately, use a strongly typed engine type to ensure we are operating on a list of values known at compile time.